### PR TITLE
[Parakeet] add output_attention_mask

### PR DESCRIPTION
**Pull request by eustlb**
# What does this PR do?

Parakeet encoder modified the sequence length, making that the input attention_mask cannot be directly used to know which embeddings positions correspond to padding. 

Here we do it differently than the actual external approach like we have on mimi that consists to retreive the correct mask:
```python
attention_mask = mimi_model.get_audio_codes_mask(inputs.attention_mask)
```
and that recomputes as mask that is already known in the forward and that can simply be returned along with the outputs.

### CODE REVIEW

File: src/transformers/models/parakeet/modeling_parakeet.py [hunk 1]
```diff
@@ -567,7 +581,9 @@ def forward(
                     **kwargs,
                 )
 
-        return BaseModelOutput(last_hidden_state=hidden_states)
+        return ParakeetEncoderModelOutput(
+            last_hidden_state=hidden_states, attention_mask=output_mask.int() if output_attention_mask else None
```
Comment (nithinraok) [inline src/transformers/models/parakeet/modeling_parakeet.py:585]: can we return lengths directly instead or along with attention_mask

Comment (eustlb) [reply src/transformers/models/parakeet/modeling_parakeet.py:585]: I'd rather keep an explicit approach here for potiential future usage of the same with left padding, you can retreive lengths by doing `attention_mask.sum(-1)`

Comment (nithinraok) [reply src/transformers/models/parakeet/modeling_parakeet.py:585]: sounds good.

**Review by ArthurZucker — APPROVED** [review]
Okay sounds good !
